sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -3192,6 +3192,19 @@ class SQLQuerySuite extends QueryTest with SharedSparkSession {
       checkAnswer(df3, Array(Row(new java.math.BigDecimal("0.100000000000000000000000100"))))
     }
   }
+
+  test("SPARK-29152: Simplify NOT(IsNull(x)) and NOT(IsNotNull(x))") {
```

Comment:
nit: I personally think  the `SimplifyConditionalSuite` test is enough for this fix.

Reply:
> nit: I personally think the `SimplifyConditionalSuite` test is enough for this fix.

Yeah, seems end-to-end test in SQLQuerySuit is redundant